src/main/java/digital/slovensko/autogram/core/Autogram.java

```diff
@@ -19,23 +19,28 @@
 
 public class Autogram {
     private final UI ui;
+    private final Settings settings;
     /** Current batch, should be null if no batch was started yet */
     private Batch batch = null;
     private final DriverDetector driverDetector;
     private final boolean shouldDisplayVisualizationError;
     private final Integer slotId;
     private final TSPSource tspSource;
+    private final PasswordManager passwordManager;
 
     public Autogram(UI ui, boolean shouldDisplayVisualizationError , DriverDetector driverDetector, TSPSource tspSource) {
         this(ui, shouldDisplayVisualizationError, driverDetector, -1, tspSource);
     }
 
-    public Autogram(UI ui, boolean shouldDisplayVisualizationError , DriverDetector driverDetector, Integer slotId, TSPSource tspSource) {
+    public Autogram(UI ui, boolean shouldDisplayVisualizationError, DriverDetector driverDetector, Integer slotId, TSPSource tspSource) {
         this.ui = ui;
+        this.settings = new DefaultSettings();
+        this.settings.setSlotId(slotId); // TODO pull out
         this.driverDetector = driverDetector;
         this.slotId = slotId;
         this.shouldDisplayVisualizationError = shouldDisplayVisualizationError;
         this.tspSource = tspSource;
+        this.passwordManager = new PasswordManager(ui, this.settings);
```

Comment:
Tu uz password manager dostava svoje settings. Aby to boli "uzke" rozhrania, tak dostava ako parameter interface `PasswordManagerSettings`, kde su len pre neho relevantne nastavenia (ziadne settery). Ocakavanie je, ze "horny" settings bude implementovat vsetky potrebne "slize" settings a posielat ich dole ako cely settings.